### ISSUE TITLE
bug: `components[].methods[].parameters` is always empty in `docs-json` output

### ISSUE THREAD

**Issue by marchbox** [ionic-team/stencil#4394]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.3

### Current Behavior

Consider this code:

```ts
/**
 * Sets something active.
 * @param {boolean} active - Whether to set something active.
 */
@Method()
async setActive(active: boolean) {
  // ...
}
```

And add `docs-json` as a output target in stencil.config.ts.

In the generated JSON file, the method’s `parameters` property is always an empty array, with and without the `@param` JsDocs.

Also the `{boolean}` part in the JsDoc is stripped out of the `docsTags.text` property, so it’s just:

```json
"docsTags": [
  {
    "name": "param",
    "text": "active - Whether to set something active."
  }
]
```

Basically there's no way to get the full documentation of parameters from neither `docs-json` nor `docs-custom` output targets.

### Expected Behavior

`parameters` property should be:

```json
"parameters": [
  {
    "name": "active",
    "type": "boolean",
    "docs": "Whether to set something active."
  }
]
```

And the `docsTags` property should be:

```json
"docsTags": [
  {
    "name": "param",
    "text": "{boolean} active - Whether to set something active."
  }
]
```

### Steps to Reproduce

1. Add a method to a component with arguments (if not already have)
2. Add `docs-json` as a output target to stencil.config.ts
3. Run `stencil build --docs`
4. See the result in the generated JSON file

### Code Reproduction URL

https://github.com/marchbox/stencil-bug-demo

### Additional Information

The code reproduction repo is using Stencil 3.0.0, but the issue also exists.

**Comment by alicewriteswrongs** [comment]
Hey @marchbox, thanks for filing this issue and for providing a reproduction! I can confirm that at present Stencil doesn't populate that field. If you take a look [here](https://github.com/ionic-team/stencil/blob/93b5cdca46d177ab8a6ae7fb2ef43c19aab00a39/src/compiler/docs/generate-doc-data.ts#L232-L232) you'll notice that some time in the misty past it was marked `// TODO`... 🧐

In any case, I'm going to label this for ingestion into our internal backlog and prioritization.

Thanks again for pointing and the issue for providing a reproduction! It really helps us triage issues 😊

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [Stencil 4.9.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.9.0).
